### PR TITLE
Unified the two number type systems

### PR DESCRIPTION
**Pull request by marcusnaslund**
Based on the discussion in, and fixes, #1143 

- `Short, Int, Long` are now of fixed size (16, 32, 64 bits respectively)
- `LLong` is as before at least 64 bits (as per `long long` in the C99 standard).
- `Octet` type removed
- `PtrDiff` type removed
- `even()`, `odd()` removed from `LLong` (implemented using `modulo` in derived classes in `math.ooc`)

### CODE REVIEW

**Comment by marcusnaslund** [comment]
This was so easy it was scary. Peer review @sebastianbaginski **and** @fredrikbryntesson ?

**Comment by sebastianbaginski** [comment]
How can you make two PRs so big after 20 minutes ? Are you a wizard

**Comment by marcusnaslund** [comment]
> Are you a wizard

Yes.

File: source/draw/gpu/opengl/backend/GLFence.ooc [hunk 1]
```diff
@@ -22,7 +22,7 @@ version(!gpuOff) {
 GLFence: abstract class {
 	_backend: Pointer = null
 	init: func
-	clientWait: abstract func (timeout: UInt64 = ULLONG_MAX) -> Bool
+	clientWait: abstract func (timeout: ULong = ULLONG_MAX) -> Bool
```
Comment (sebastianbaginski) [inline source/draw/gpu/opengl/backend/GLFence.ooc]: Can you change this to `ULONG_MAX` ?

File: source/draw/gpu/opengl/backend/gles3/Gles3Fence.ooc [hunk 2]
```diff
@@ -29,7 +29,7 @@ Gles3Fence: class extends GLFence {
 		version(debugGL) { validateEnd("Fence free") }
 		super()
 	}
-	clientWait: override func (timeout: UInt64 = ULLONG_MAX) -> Bool {
+	clientWait: override func (timeout: ULong = ULLONG_MAX) -> Bool {
```
Comment (sebastianbaginski) [inline source/draw/gpu/opengl/backend/gles3/Gles3Fence.ooc]: Same here `timeout: ULong = ULONG_MAX`

**Comment by fredrikbryntesson** [comment]
Outstanding

File: source/sdk/os/win32.ooc [hunk 3]
```diff
@@ -26,7 +26,7 @@ version(windows) {
 		buf setLength(len)
 
 		// rip away trailing CR LF TAB SPACES etc.
-		while ((len > 0) && (buf[len - 1] as Octet < 32)) len -= 1
+		while ((len > 0) && (buf[len - 1] as SByte < 32)) len -= 1
```
Comment (marcusnaslund) [inline source/sdk/os/win32.ooc]: Oops

**Pull request merged**